### DISCUSSION TITLE
Chapter 16 OpenId code issues

### DISCUSSION THREAD

**Discussion by SvdSinner** [PeterHimschoot/microsoft-blazor-book-3#6] [General]
``The Blazor server project in chapter 16 has a few issues:

- The current .NET 6.0 Blazor Server template no longer uses Startup.cs
- There is no mention in the book to go to launchSettings.json and change the port to 5000/5001.
- After adapting past the two above issues,  and following the steps, the RedirectToLogin.razor component throw an exception on launch that I can't get passed: 
```
-Microsoft.AspNetCore.Components.NavigationException
  HResult=0x80131500
  Message=Exception of type 'Microsoft.AspNetCore.Components.NavigationException' was thrown.
  Source=Microsoft.AspNetCore.Components.Server
  StackTrace:
   at Microsoft.AspNetCore.Components.Server.Circuits.RemoteNavigationManager.NavigateToCore(String uri, NavigationOptions options)
   at Microsoft.AspNetCore.Components.NavigationManager.NavigateToCore(String uri, Boolean forceLoad)
   at Microsoft.AspNetCore.Components.NavigationManager.NavigateTo(String uri, Boolean forceLoad, Boolean replace)
   at Blazor.Shared.RedirectToLogin.OnInitialized() in C:\Users\gwilson\source\repos\IdentityProvider\Blazor\Shared\RedirectToLogin.razor:line 10
   at Microsoft.AspNetCore.Components.ComponentBase.<RunInitAndSetParametersAsync>d__20.MoveNext()
```
 I've gone through the files I made versus the files you have here, and can't find anything obvious as to why the exception happens when the project is created from the modern .NET 6.0 Blazor Server template

**Comment by PeterHimschoot** [comment]
Hi Greg,

This is great feedback.

Concerning `Startup`, I drafted the book using the preview version, and at the last possible moment Microsoft changed the template. :)
At the same time, they changed the project template to choose a random port number, not the default 5001.

About the `NavigationException` you keep getting, simply press `F5` to continue running the code. Somehow this method always throws this exception, you can disable VS stopping on it using Debug->Windows->Exception settings, or run without the debugger...

I am currently updating the sample code to use the latest templates, I will get chapter 16 in your hands ASAP.

**Comment by PeterHimschoot** [comment]
Hi Greg,

I have uploaded an updated version of the project to GitHub, so don't forget to pull this version if you want to compare. There are also a bunch of notes included that hopefully will make the experience smooth.

**Comment by SvdSinner** [comment]
Thank you!
May I ask you the question that has got me digging deeply into security,
identity, authentication and authorization?   I'm spearheading an internal
Blazor application at my company where we use WsFederation servers for our
web apps.  The desired behavior is that our internal users who are already
l logged into AD will see no login screen and be properly identified (Basic
SSO behavior)
I can make this happen by hand if I ignore all of the built in Identity,
Authentication and Authorization middleware provided by Asp.Net Core, but I
would prefer to use the built in middleware.

What general strategy do I have to use? Can I use Identity for user
information, or is Identity completely built on the paradigm of log in/log
out screens?  If I can't or shouldn't use Identity, how do I get
WsFederation to play nicely with Authentication and Authorization
middleware?

I'm not needing you to write code for me, I'm just struggling with the
general strategy.  Information on using WsFederation for SSO in Blazor is
virtually non-existant and my unanswered question on StackOverflow is one
of the top Google results on the topic.

Greg Wilson

P.S.  I've really enjoyed your book.  I'm a huge believer in diving into
how technologies (like Blazor and Asp.Net Core) work before architecting
applications that use them.
